manager/src/main/java/com/redhat/service/smartevents/manager/ProcessorServiceImpl.java

```diff
@@ -83,15 +83,31 @@ public Processor getProcessor(String bridgeId, String processorId, String custom
     @Override
     @Transactional
     public Processor createProcessor(String bridgeId, String customerId, String owner, ProcessorRequest processorRequest) {
-        /* We cannot deploy Processors to a Bridge that is not Available */
+        // We cannot deploy Processors to a Bridge that is not available. This throws an Exception if the Bridge is not READY.
         Bridge bridge = bridgesService.getReadyBridge(bridgeId, customerId);
 
+        return doCreateProcessor(bridge, customerId, owner, processorRequest.getType(), processorRequest, false);
+    }
+
+    @Override
+    @Transactional
+    public Processor createErrorHandlingProcessor(String bridgeId, String customerId, String owner, ProcessorRequest processorRequest) {
+        Bridge bridge = bridgesService.getBridge(bridgeId, customerId);
+
+        ProcessorType processorType = processorRequest.getType();
+        if (processorType != ProcessorType.SINK) {
+            throw new BadRequestException("Unable to create a SOURCE Processor for Error Handling.");
+        }
```

Comment:
Is this correct, don't we have now a specific error handling processor type?
Btw the error message seems wrong to me

Reply:
updated :+1: